lib/modules/datasource/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  DescribeAddonVersionsCommand,
+  DescribeAddonVersionsCommandInput,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
+import { cache } from '../../../util/cache/package/decorator';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type { EKSAddonsFilter } from './types';
+
+export class AwsEKSAddonDataSource extends Datasource {
+  static readonly id = 'aws-eks-addon';
+
+  override readonly caching = true;
+
+  constructor() {
+    super(AwsEKSAddonDataSource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${AwsEKSAddonDataSource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `getReleases:${packageName}`,
+  })
+  async getReleases({
+    packageName: serializedFilter,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const filter: EKSAddonsFilter = JSON.parse(serializedFilter);
+    const eksClient = this.getEKSClient(filter);
+
+    const cmd = new DescribeAddonVersionsCommand(
+      this.getDescribeAddonsRequest(filter),
+    );
+    const response = await eksClient.send(cmd);
+    const addons = response.addons ?? [];
+    return {
+      releases: addons
+        .flatMap((addon) => addon.addonVersions)
+        .map((versionInfo) => ({
+          version: versionInfo!.addonVersion!,
```

Comment:
why do we need all those `!`?

Reply:
AWS javascript SDK makes everything nullable even for basic fields that should always exist. I added an extra `filter` to safeguard this (and updated unit tests). Hope it looks better to you :)

Reply:
do an additional map and filter after, so no need for any `!`

Reply:
I refactored it a bit, hope it looks better to you.